internal/implementations/secret/secret_test.go

```diff
@@ -0,0 +1,64 @@
+package secret_test
```

Comment:
I realized after looking at PR #139 that I need to add a test suite file for this. Just waiting for @pleshakov's response on this question: https://github.com/nginxinc/nginx-kubernetes-gateway/pull/139#discussion_r919305747 before adding it.

Reply:
I updated all the package names to implementation and added a test suite for secret.